testsuite/features/step_definitions/common_steps.rb

```diff
@@ -371,22 +371,71 @@
 
 # setup wizard
 
-When(/^I make the credentials primary$/) do
-  raise 'xpath: i.fa-star-o not found' unless find('i.fa-star-o').click
+Then(/^HTTP proxy verification should have succeeded$/) do
+  raise 'xpath: i.text-success not found' unless find('i.text-success', wait: DEFAULT_TIMEOUT)
 end
 
-When(/^I delete the primary credentials$/) do
-  raise 'xpath: i.fa-trash-o not found' unless find('i.fa-trash-o', match: :first).click
-  step 'I click on "Delete"'
+When(/^I enter the address of the HTTP proxy as "([^"]*)"/) do |hostname|
+  step %(I enter "#{$server_http_proxy}" as "#{hostname}")
+end
+
+When(/^I ask to add new credentials$/) do
+  raise 'xpath: i.fa-plus-circle not found' unless find('i.fa-plus-circle').click
+end
+
+When(/^I enter the SCC credentials$/) do
+  scc_username = ENV['SCC_CREDENTIALS'].split('|')[0]
+  scc_password = ENV['SCC_CREDENTIALS'].split('|')[1]
+  steps %(
+    And I enter "#{scc_username}" as "edit-user"
+    And I enter "#{scc_password}" as "edit-password"
+  )
+end
+
+Then(/^the SCC credentials should be valid$/) do
+  scc_username = ENV['SCC_CREDENTIALS'].split('|')[0]
+  scc_password = ENV['SCC_CREDENTIALS'].split('|')[1]
+  within(:xpath, "//h3[contains(text(), '#{scc_username}')]/../..") do
+    raise 'Valid icon not found' unless find('i.text-success', wait: DEFAULT_TIMEOUT)
+  end
+end
+
+Then(/^the credentials for "([^"]*)" should be invalid$/) do |user|
+  within(:xpath, "//h3[contains(text(), '#{user}')]/../..") do
+    raise 'Invalid icon not found' unless find('i.text-danger', wait: DEFAULT_TIMEOUT)
+  end
+end
+
+When(/^I make the credentials for "([^"]*)" primary$/) do |user|
+  within(:xpath, "//h3[contains(text(), '#{user}')]/../..") do
+    raise 'Star icon not found' unless find('i.fa-star-o').click
+  end
 end
 
-When(/^I view the primary subscription list$/) do
-  raise 'xpath: i.fa-th-list not found' unless find('i.fa-th-list', match: :first).click
+Then(/^the credentials for "([^"]*)" should be primary$/) do |user|
+  within(:xpath, "//h3[contains(text(), '#{user}')]/../..") do
+    raise 'Star icon not selected' unless find('i.fa-star')
+  end
+end
+
+When(/^I wait for the trash icon to appear for "([^"]*)"$/) do |user|
+  within(:xpath, "//h3[contains(text(), '#{user}')]/../..") do
+    repeat_until_timeout(message: 'Trash icon is still greyed out') do
+      break unless find('i.fa-trash-o')[:style].include? "not-allowed"
+      sleep 1
```

Comment:
We don't need to add the sleep method, the find will already have its default timeout.
Even, I would avoid to use the repeat_until_timeout method instead use `find('i.fa-trash-o', wait: your_timeout)[:style].include? "not-allowed"`

Reply:
ok